utils/sl/prereq_tools/base.py

```diff
@@ -366,28 +367,14 @@ def get(self, subdir, **kw):
 *********************** ERROR ************************\n""" % comp)
             raise DownloadFailure(self.url, subdir)
 
-        commands = ['git clone %s %s' % (self.url, subdir)]
-        if not RUNNER.run_commands(commands):
+        if (passed_commit_sha and not branch):
+            commands = ['git clone %s --branch %s --single-branch %s' % (self.url,
```

Comment:
I think you also want to use `--depth 1` to be the most efficient.

Reply:
Done.

The submodules aren't using a shallow clone, although that doesn't seem to be as easy to setup.